rocAL/rocAL/include/api/rocal_api_augmentation.h

```diff
@@ -385,42 +476,56 @@ extern "C"  RocalTensor  ROCAL_API_CALL rocalRain(RocalContext context, RocalTen
+                                                     float rain_transparency,
+                                                     bool is_output,
+                                                     RocalTensorLayout rocal_tensor_output_layout = ROCAL_NHWC,
+                                                     RocalTensorOutputType rocal_tensor_output_datatype = ROCAL_UINT8);
 
 /// \param context
 /// \param input
 /// \param is_output
 /// \param adjustment
 /// \return
-extern "C"  RocalTensor  ROCAL_API_CALL rocalColorTemp(RocalContext context, RocalTensor input, bool is_output,
-                                                        RocalIntParam adjustment = NULL);
+extern "C" RocalTensor ROCAL_API_CALL rocalColorTemp(RocalContext context, RocalTensor input,
+                                                     bool is_output,
+                                                     RocalIntParam adjustment = NULL,
+                                                     RocalTensorLayout rocal_tensor_output_layout = ROCAL_NHWC,
+                                                     RocalTensorOutputType rocal_tensor_output_datatype = ROCAL_UINT8);
 
 /// \param context
 /// \param input
 /// \param adjustment
 /// \param is_output
 /// \return
-extern "C"  RocalTensor  ROCAL_API_CALL rocalColorTempFixed(RocalContext context, RocalTensor input, int adjustment, bool is_output);
+extern "C" RocalTensor ROCAL_API_CALL rocalColorTempFixed(RocalContext context, RocalTensor input,
+                                                          int adjustment, bool is_output,
+                                                          RocalTensorLayout rocal_tensor_output_layout = ROCAL_NHWC,
+                                                          RocalTensorOutputType rocal_tensor_output_datatype = ROCAL_UINT8);
 
 /// \param context
 /// \param input
 /// \param is_output
 /// \param fog_value
 /// \return
-extern "C"  RocalTensor  ROCAL_API_CALL rocalFog(RocalContext context, RocalTensor input, bool is_output,
-                                                RocalFloatParam fog_value = NULL);
+extern "C" RocalTensor ROCAL_API_CALL rocalFog(RocalContext context, RocalTensor input,
+                                               bool is_output,
+                                               RocalFloatParam fog_value = NULL,
+                                               RocalTensorLayout rocal_tensor_output_layout = ROCAL_NHWC,
+                                               RocalTensorOutputType rocal_tensor_output_datatype = ROCAL_UINT8);
 
 /// \param context
 /// \param input
 /// \param fog_value
 /// \param is_output
 /// \return
-extern "C"  RocalTensor  ROCAL_API_CALL rocalFogFixed(RocalContext context, RocalTensor input, float fog_value, bool is_output);
+extern "C" RocalTensor ROCAL_API_CALL rocalFogFixed(RocalContext context, RocalTensor input,
+                                                    float fog_value, bool is_output,
```

Comment:
Is it possible to use templates to combine the fixed and normal api's and handle them internally based on the type?
since this api's are only going to increase, it might be hard to maintain both versions
Please check if we can use templates and handle them inside the function call for fixed/random generation

Reply:
@sampath1117 We can make this change in future.
With the development changes for external source all API must support tensor. It will be better to change then.
@shobana-mcw Please note